src/transformers/portable-text-transformer/portable-text-transformer.ts

```diff
@@ -153,49 +155,36 @@ const mergeRowsAndCells = (itemsToMerge: IPortableTextItem[]): IPortableTextItem
     return mergedItems;
 }
 
-const mergeCellsAndBlocks = (itemsToMerge: IPortableTextItem[]): IPortableTextItem[] => {
-    const mergedItems = itemsToMerge.reduce<IPortableTextItem[]>((mergedItems, item, index, array) => {
-        if (item._type === 'cell') {
-            for (let i = 1; i <= item.childBlocksCount; i++) {
-                item.content.push(array[index + i] as IPortableTextBlock);
-            }
-            array.splice(index, item.childBlocksCount);
-            mergedItems.push(item);
-        } else {
-            mergedItems.push(item);
-        }
-
-        return mergedItems;
-    }, [])
-
-    return mergedItems;
-}
-
-const composeAndMerge = compose(mergeTablesAndRows, mergeRowsAndCells, mergeCellsAndBlocks, mergeBlocksAndSpans, mergeSpansAndMarks);
+const composeAndMerge = compose(mergeTablesAndRows, mergeRowsAndCells, mergeBlocksAndSpans, mergeSpansAndMarks);
 
-const flatten = (nodes: IDomNode[], depth = 0, listType?: ListType): IPortableTextItem[] => {
+const flatten = (nodes: IDomNode[], depth = 0, lastListElement?: IDomHtmlNode, listType?: ListType): IPortableTextItem[] => {
     return nodes.flatMap((node: IDomNode) => {
         let children: IDomNode[] = [];
         let transformedChildren: IPortableTextItem[] = [];
         let currentListType = listType;
-        let listDepthIncrement = 0;
         const finishedItems: IPortableTextItem[] = [];
 
         if (isElement(node)) {
-            children = node.children;
+            children = node.tagName === 'td' ? [] : node.children;
```

Comment:
This small exception for table cells indicates that a more natural approach to the algorithm would've been a recursive algorithm (basically what you do now for cells).